git-extra/git-update

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# Compares the currently installed Git for Windows against latest available release.
+# If versions differ, the bit matched installer is downloaded and run when confirmation
+# to do so is given.
+
+git_update () {
+    yn="$1"
+    case $yn in
+        -\? | --?\?) echo "Usage: git-update [options]";
+        echo "Options:";
+        echo "  -y, --yes       Automatic yes to download and install prompt";
+        return;;
+    esac
+    bit=$(ls --color=never -d /mingw* | sed 's/\/mingw//')
+    releases=$(curl --silent https://api.github.com/repos/git-for-windows/git/releases/latest)
+    latest=$(echo "$releases" | grep '"tag_name": "v' | sed -E 's/.*"tag_name": "v([^"]*).*/\1/')
+    version=$(git --version | sed "s/git version //")
+    echo "Git for Windows $version (${bit}bit)"
+    if [ "$latest" != "$version" ]
+    then
+        echo "Update $latest is available"
+        download=$(echo "$releases" | grep '"browser_download_url": "' | grep "$bit\-bit\.exe" | sed -E 's/.*": "([^"]*).*/\1/')
+        filename=$(echo "$download" | sed -E 's/.*\/([^\/]*)$/\1/')
+        installer=$(mktemp -t gfw-install-XXXXXXXX.exe)
+        case $yn in
+            -[Yy]* | --[Yy]*) yn="y"; echo "Downloading $filename";;
+            *) read -p "Download $filename and install [N/y]? " yn;;
+        esac
+        case $yn in
+            [Yy]* ) curl -# -L -o $installer $download; start "" "$installer" /SILENT; ps | grep bash | awk '{print "kill -9 " $2 ";" }' | sh;;
```

Comment:
@larsxschneider The following command seems to isolate the published sha256 sum, where ${bit} is the calculated bitness of your current installation.

SHA256=$(curl --silent --location https://github.com/git-for-windows/git/releases/latest | awk '/<td>/' | sed 'N;s/\n/ /' | grep "${bit}-bit\.exe" | sed -E 's/.*<td>([^<]+)<\/td>/\1/')

i.e.
\# curl --silent --location https://github.com/git-for-windows/git/releases/latest | awk '/<td>/' | sed 'N;s/\n/ /' | grep "64-bit\.exe" | sed -E 's/.*<td>([^<]+)<\/td>/\1/'

Problem is, wouldn't parsing the expected sum from HTML be less reliable than curl successfully downloading and writing an uncorrupted binary to disk?

If the parse ever goes wrong the correct sum won't be stored in $SHA256 and the update script will incorrectly abort.